src/main/java/jnr/ffi/Platform.java

```diff
@@ -500,32 +511,81 @@ public String mapLibraryName(String libName) {
             }
             return "lib" + libName + ".dylib";
         }
-        
-        @Override
-        public String getName() {
-            return "Darwin";
-        }
 
     }
+
     /**
      * A {@link Platform} subclass representing the Linux operating system.
      */
     static final class Linux extends Supported {
 
+        // represents a valid library file that matches the search
+        private static class Match implements Comparable<Match> {
+            String path; // absolute path of library file
+            int[] version; // version of library, empty if no version specified
+            boolean isCustom; // if path is from a custom searchPath specified by user and not default path
+
+            @Override
+            public int compareTo(Match o) { // for Collections.sort() to work
+                return compareVersions(o.version, this.version);
+            }
+        }
+
         public Linux() {
             super(OS.LINUX);
         }
 
         @Override
-        public String locateLibrary(final String libName, List<String> libraryPaths) {
+        public String locateLibrary(String libName, List<String> libraryPaths) {
+            return locateLibrary(libName, libraryPaths, null);
+        }
+
+        @Override
+        public String locateLibrary(final String libName, List<String> libraryPaths,
```

Comment:
Should this `Map` be a Set?

Reply:
Yes but the rest of the APIs all use `Map<LibraryOption, Object>` when taking in the `options`, I'm simply following the standard. This makes things consistent and easy for any future changes too so that we don't have to convert from `Map` to `Set`, but yeah you are indeed right, `Set` definitely makes more sense overall.